apps/hash-frontend/src/pages/flows.page/flow-definition/run-flow-modal.tsx

```diff
@@ -0,0 +1,258 @@
+import { IconButton } from "@hashintel/design-system";
+import { typedValues } from "@local/advanced-types/typed-entries";
+import type {
+  FlowDefinition,
+  FlowTrigger,
+  OutputDefinition,
+  PayloadKind,
+  StepOutput,
+} from "@local/hash-isomorphic-utils/flows/types";
+import type { OwnedById } from "@local/hash-subgraph";
+import { Box, Stack, Typography } from "@mui/material";
+import type { PropsWithChildren } from "react";
+import { useState } from "react";
+
+import { XMarkRegularIcon } from "../../../shared/icons/x-mark-regular-icon";
+import { Button } from "../../../shared/ui/button";
+import { Modal } from "../../../shared/ui/modal";
+import { useAuthenticatedUser } from "../../shared/auth-info-context";
+import { ManualTriggerInput } from "./run-flow-modal/manual-trigger-input";
+import type { LocalPayload, LocalPayloadKind } from "./run-flow-modal/types";
+
+type FormState = {
+  [outputName: string]: {
+    outputName: string;
+    payload: LocalPayload;
+  };
+};
+
+const InputWrapper = ({
+  children,
+  required,
+  label,
+}: PropsWithChildren<{ required: boolean; label: string }>) => (
+  <Box mb={2.5}>
+    <Typography
+      component="label"
+      variant="smallTextLabels"
+      sx={{
+        color: ({ palette }) => palette.gray[70],
+        fontWeight: 500,
+        lineHeight: 1.5,
+      }}
+    >
+      {label}
+      {required ? "*" : ""}
+      <Box>{children}</Box>
+    </Typography>
+  </Box>
+);
+
+type UnsupportedPayloadKind = Exclude<PayloadKind, LocalPayloadKind>;
+
+const unsupportedPayloadKinds: UnsupportedPayloadKind[] = [
+  "EntityType",
```

Comment:
We have a type selector in the form already for `VersionedUrl`, is it difficult to add support for `EntityType` in addition to `VersionedUrl`? If not we could add support for it now as well.

Reply:
No, it wouldn't be. We might be better off just picking one though – maybe `VersionedUrl` since the `EntityType` can be resolved from it if necessary in any action, whereas `EntityType` provides an additional burden on the source to have already resolved it, which it might not need to?

Reply:
Cool, I'll remove it as a drive-by in the next PR